### PR TITLE
Drop support for nodev4.

### LINKED ISSUE #86: Drop support for node.js 4

### PR DESCRIPTION
**Pull request by kolodny**
## Node 4 goes EOL 4/30/18

Fixes #86

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -73,38 +67,6 @@ workflows:
               only: master
     jobs: *workflow_jobs
 jobs:
-  node4:
-    docker:
-      - image: 'node:4'
-        user: node
-    steps: &unit_tests_steps
```
Comment (stephenplusplus) [inline .circleci/config.yml]: We have to move this section to the next-lowest version supported. This is how the nightly tests work, I believe.

@alexander-fenster do we want to make this change one-off here (in other words, do we want this PR), or can the generator do it across all of our projects at once?

Comment (alexander-fenster) [reply .circleci/config.yml]: Both are fine (the generator generally refuses to make a change if it's already made, so nothing bad will happen if the node4 task is already deleted). But do we stop testing on node4 right now already? 

My understanding is that we want to wait at least for 4/30 when Node 4 is officially EOL. I'd like to add Node 10 into the same change so that we'd have one change across all repos, and not two.

In other words, I'd probably like to wait for 4/30 and then I'll make the change across all repos to disable node4 and add node10.

Comment (alexander-fenster) [reply .circleci/config.yml]: And, for sure, removal of this section is wrong as it's referenced from other jobs.

**Comment by alexander-fenster** [comment]
@kolodny why do you want to remove node4 now? It's technically alive now (till 4/30).

**Comment by kolodny** [comment]
@alexander-fenster My mistake, I thought EOL was 4/1 I'll close this PR for now and reopen after that time
